test/domain/services/notification_service_test.dart

```diff
@@ -0,0 +1,279 @@
+                jsonDecode(notification.payload!) as Map,
+              );
+              if (payload['groupId'] == targetGroupId) {
+                notificationsToCancel.add(notification.id!);
+                await mockPlugin.cancel(notification.id!);
+              }
+            } catch (e) {
+              continue;
+            }
+          }
+        }
+
+        // Verify only notification with matching groupId was cancelled
+        expect(notificationsToCancel, equals([1]));
+        verify(mockPlugin.cancel(1)).called(1);
+        verifyNever(mockPlugin.cancel(2)); // No groupId, not cancelled
+        verifyNever(mockPlugin.cancel(3)); // Null payload, not processed
+      });
+
+      test('should handle empty active notifications list', () async {
+        // Arrange: No active notifications
+        when(mockPlugin.getActiveNotifications()).thenAnswer((_) async => []);
+
+        // Test the logic
+        final activeNotifications = await mockPlugin.getActiveNotifications();
+        const targetGroupId = 'target-group';
+        final notificationsToCancel = <int>[];
+
+        for (final notification in activeNotifications) {
+          if (notification.payload != null) {
+            try {
+              final payload = Map<String, dynamic>.from(
+                jsonDecode(notification.payload!) as Map,
+              );
+              if (payload['groupId'] == targetGroupId) {
+                notificationsToCancel.add(notification.id!);
+                await mockPlugin.cancel(notification.id!);
+              }
+            } catch (e) {
+              continue;
+            }
+          }
+        }
+
+        // Verify no notifications were cancelled
+        expect(notificationsToCancel, isEmpty);
+        verify(mockPlugin.getActiveNotifications()).called(1);
+        verifyNever(mockPlugin.cancel(any));
+      });
+    });
+
+    group('actual service calls', () {
+      test('cancelNotificationsByGroup should handle uninitialized state', () async {
+        // This tests the actual method call, which should return early if not initialized
+        expect(() async {
+          await NotificationService.cancelNotificationsByGroup('test-group');
+        }, returnsNormally);
+      });
+
+      test('should handle edge cases with groupId parameter', () async {
+        expect(() async {
+          await NotificationService.cancelNotificationsByGroup('');
+          await NotificationService.cancelNotificationsByGroup('  ');
+          await NotificationService.cancelNotificationsByGroup('group-with-special-chars-"quotes"');
+        }, returnsNormally);
+      });
+
+      test('other service methods should handle uninitialized state', () async {
+        expect(() async {
+          await NotificationService.cancelNotification(-1);
+          await NotificationService.cancelAllNotifications();
+        }, returnsNormally);
+      });
+    });
+
+    group('payload structure validation', () {
```

Comment:
Please remove this tests, which is just testing jsonEncode, not the notification service class